tests/app/core/models/session-model.spec.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  assertSessionDataIntegrity,
+  SessionData,
+  SESSION_DATA_INCORRECTLY_DEFINED,
+  SESSION_DATA_UNAVAILABLE,
+  NO_ACCESSIBLE_PHASES,
+} from '../../../../src/app/core/models/session.model';
+import { Phase } from '../../../../src/app/core/services/phase.service';
+import { of } from 'rxjs';
+
+const validSessionData: SessionData = {
+  openPhases: [
+    Phase.phaseBugReporting,
+    Phase.phaseTeamResponse,
+    Phase.phaseTesterResponse,
+    Phase.phaseModeration,
+  ],
+  phaseBugReporting: 'bugreporting',
+  phaseTeamResponse: 'pe-results',
+  phaseTesterResponse: 'testerresponse',
+  phaseModeration: 'pe-evaluation',
+};
+
+describe('Session Model', () => {
+  describe('assertSessionDataIntegrity', () => {
+    it('should throw error on unavailable session', () => {
+      of(undefined)
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(new Error(SESSION_DATA_UNAVAILABLE)),
```

Comment:
Perhaps instead of expecting the fixed error string, a better approach might be to expect an error itself so that the tests work regardless of the string attached to the error. There is an example here https://github.com/CATcher-org/CATcher/pull/560#discussion_r566607391

This way I think can undo the changes done to `session.model.ts`  to remove the circular dependency error and maybe we can open an issue to discuss shifting the `Phase` `enum` to its own file so that it can be imported into the `session.model.ts` to be used in defining the `SessionData`

Reply:
Hmm..I actually find it helpful to verify the error message itself - it gives us assurance that the user sees the right error?

@ptvrajsk  do you see any downside to verifying the message? (it's stored in a constant now - we won't have to change the tests when we edit the message)


> maybe we can open an issue to discuss shifting the Phase enum to its own file so that it can be imported into the session.model.ts to be used in defining the SessionData

I agree with this :+1: @dingyuchen can you setup an issue for this?

Reply:
> maybe we can open an issue to discuss shifting the Phase enum to its own file so that it can be imported into the session.model.ts to be used in defining the SessionData

Ah yes that makes more sense. Will do.

> For this PR maybe can leave the main SessionData interface alone?

I think that we don't have to go so far in order to avoid this warning since the circular dependency is only in the files, not any class/object or type definition. Furthermore I think enforcing the keys of SessionData to be consistent is more pertinent as well.

Reply:
> Furthermore I think enforcing the keys of SessionData to be consistent is more pertinent as well.

This is indeed a valid change, but let's do it in a separate PR as it isn't closely related to this PR's purpose